From b159eac478fc451407d9e113e976a170b919ee3c Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Wed, 23 Feb 2022 16:23:36 -0800
Subject: [PATCH] imx8mp_var_dart: Allow wifi / BT vendor init script

Whitelist wifi and bt script to execute under vendor init

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 prebuilts/api/30.0/public/property.te | 22 ++++------------------
 public/property.te                    | 22 ++++------------------
 2 files changed, 8 insertions(+), 36 deletions(-)

diff --git a/prebuilts/api/30.0/public/property.te b/prebuilts/api/30.0/public/property.te
index 025ceaddc..ab57816cc 100644
--- a/prebuilts/api/30.0/public/property.te
+++ b/prebuilts/api/30.0/public/property.te
@@ -421,6 +421,8 @@ compatible_property_only(`
     -nfc_prop
     -powerctl_prop
     -radio_prop
+    -wifi_prop
+    -bluetooth_prop
   }:property_service set;
 
   neverallow {
@@ -453,15 +455,6 @@ compatible_property_only(`
     radio_prop
   }:property_service set;
 
-  neverallow {
-    domain
-    -coredomain
-    -bluetooth
-    -hal_bluetooth_server
-  } {
-    bluetooth_prop
-  }:property_service set;
-
   neverallow {
     domain
     -coredomain
@@ -482,15 +475,6 @@ compatible_property_only(`
     exported_camera_prop
   }:property_service set;
 
-  neverallow {
-    domain
-    -coredomain
-    -hal_wifi_server
-    -wificond
-  } {
-    wifi_prop
-  }:property_service set;
-
   neverallow {
     domain
     -coredomain
@@ -549,6 +533,7 @@ compatible_property_only(`
     -coredomain
     -bluetooth
     -hal_bluetooth_server
+    -vendor_init
   } {
     bluetooth_prop
   }:file no_rw_file_perms;
@@ -558,6 +543,7 @@ compatible_property_only(`
     -coredomain
     -hal_wifi_server
     -wificond
+    -vendor_init
   } {
     wifi_prop
   }:file no_rw_file_perms;
diff --git a/public/property.te b/public/property.te
index 025ceaddc..ab57816cc 100644
--- a/public/property.te
+++ b/public/property.te
@@ -421,6 +421,8 @@ compatible_property_only(`
     -nfc_prop
     -powerctl_prop
     -radio_prop
+    -wifi_prop
+    -bluetooth_prop
   }:property_service set;
 
   neverallow {
@@ -453,15 +455,6 @@ compatible_property_only(`
     radio_prop
   }:property_service set;
 
-  neverallow {
-    domain
-    -coredomain
-    -bluetooth
-    -hal_bluetooth_server
-  } {
-    bluetooth_prop
-  }:property_service set;
-
   neverallow {
     domain
     -coredomain
@@ -482,15 +475,6 @@ compatible_property_only(`
     exported_camera_prop
   }:property_service set;
 
-  neverallow {
-    domain
-    -coredomain
-    -hal_wifi_server
-    -wificond
-  } {
-    wifi_prop
-  }:property_service set;
-
   neverallow {
     domain
     -coredomain
@@ -549,6 +533,7 @@ compatible_property_only(`
     -coredomain
     -bluetooth
     -hal_bluetooth_server
+    -vendor_init
   } {
     bluetooth_prop
   }:file no_rw_file_perms;
@@ -558,6 +543,7 @@ compatible_property_only(`
     -coredomain
     -hal_wifi_server
     -wificond
+    -vendor_init
   } {
     wifi_prop
   }:file no_rw_file_perms;
-- 
2.25.1

